src/overlays/actors/ovl_Bg_Hidan_Dalm/z_bg_hidan_dalm.c

```diff
@@ -21,17 +29,175 @@ const ActorInit Bg_Hidan_Dalm_InitVars = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, 65336, ICHAIN_STOP),
+};
+
+Vec3f D_808865A8 = { 0, 0, 0 };
+
+#ifdef NON_MATCHING
+// regalloc
+void BgHidanDalm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanDalm* this = THIS;
+    s32 switchFlag;
+    u32 dynaUnk;
+
+    dynaUnk = 0;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK);
+    DynaPolyInfo_Alloc(&D_0600DA10, &dynaUnk);
+    // = ... & 0xFFFFFFFFFFFFFFFF; moves regalloc issues
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, dynaUnk);
+    Collider_InitTris(globalCtx, &this->unk_16C);
+    Collider_SetTris(globalCtx, &this->unk_16C, thisx, &sTrisInit, this->unk_18C);
+
+    switchFlag = thisx->params >> 8; /*
+    // this instead of the line above solves half the regalloc issues
+    switchFlag = thisx->params;
+    switchFlag = switchFlag >> 8 & 0xFFFFFFFFFFFFFFFF;
+    //*/
+    this->switchFlag = switchFlag; // can use thisx->params >> 8 without changes
+    thisx->params &= 0xFF;
+    // can use thisx->params >> 8 , seems to improve regalloc, but it changes codegen
+    if (Flags_GetSwitch(globalCtx, switchFlag & 0xFF)) {
+        Actor_Kill(thisx);
+    } else {
+        this->actionFunc = &BgHidanDalm_WaitInteraction;
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Dalm/BgHidanDalm_Init.s")
+#endif
+
+void BgHidanDalm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanDalm* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Dalm/BgHidanDalm_Destroy.s")
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyTris(globalCtx, &this->unk_16C);
+}
+
+void BgHidanDalm_WaitInteraction(BgHidanDalm* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->unk_16C.base.acFlags & 2) && !func_8008E988(globalCtx) &&
+        (player->swordAnimation == 22 || player->swordAnimation == 23)) {
+        this->unk_16C.base.acFlags &= ~(1 << 1);
+        if (this->unk_16C.list[0].body.bumperFlags & 2 || this->unk_16C.list[1].body.bumperFlags & 2) {
+            this->dyna.actor.posRot.rot.y -= 0x4000;
+        } else {
+            this->dyna.actor.posRot.rot.y += 0x4000;
+        }
+        this->dyna.actor.posRot.pos.x += 32.5f * Math_Sins(this->dyna.actor.posRot.rot.y);
+        this->dyna.actor.posRot.pos.z += 32.5f * Math_Coss(this->dyna.actor.posRot.rot.y);
+
+        func_8002DF54(globalCtx, &this->dyna.actor, 8);
+        this->dyna.actor.flags |= 1 << 4;
+        this->actionFunc = &BgHidanDalm_Shrink;
+        this->dyna.actor.bgCheckFlags &= ~(1 << 1);
+        this->dyna.actor.bgCheckFlags &= ~(1 << 3);
+        this->dyna.actor.speedXZ = 10.0f;
+        Flags_SetSwitch(globalCtx, this->switchFlag);
+        func_8002F7DC(&PLAYER->actor, NA_SE_IT_HAMMER_HIT);
```

Comment:
youve got a player temp, can you use it? 
want to avoid inline macro usages for THIS and PLAYER as much as we can

Reply:
unfortunately, no